tests/runtest.py

```diff
@@ -625,6 +635,7 @@ def run_single_case(case, flags, opts, arg):
     exec("import %s; tc = %s.TestCase()" % (case, case), globals(), _locals)
     tc = _locals['tc']
     tc.set_debug(arg.debug)
+    tc.set_compiler(arg.compiler)
```

Comment:
It might be better to have a way to set the compiler globally but it doesn't look possible in the current design because the target test program imports this. It'd be better if you have any better idea, but I'm okay with the current approach.